modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/portlet/action/test/CalculateSegmentsExperimentEstimatedDurationMVCActionCommandTest.java

```diff
@@ -0,0 +1,275 @@
+
+		SegmentsExperimentRel segmentsExperimentRel = _addSegmentsExperimentRel(
+			segmentsExperience, segmentsExperiment);
+
+		MockLiferayPortletActionResponse mockLiferayPortletActionResponse =
+			new MockLiferayPortletActionResponse();
+
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			_getMockLiferayPortletActionRequests();
+
+		mockLiferayPortletActionRequest.addParameter(
+			"segmentsExperimentId",
+			String.valueOf(segmentsExperiment.getSegmentsExperimentId()));
+
+		mockLiferayPortletActionRequest.addParameter(
+			"segmentsExperimentRels",
+			JSONUtil.put(
+				String.valueOf(
+					segmentsExperimentRel.getSegmentsExperimentRelId()),
+				"0.5"
+			).toString());
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			_mvcActionCommand, "_asahFaroBackendClient");
+
+		int estimatedDuration = 12;
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				HashMapBuilder.<String, UnsafeSupplier<String, Exception>>put(
+					StringUtil.replace(
+						"/api/1.0/experiments/{experimentId}" +
+							"/estimated-days-duration",
+						"{experimentId}",
+						segmentsExperiment.getSegmentsExperimentKey()),
+					() -> String.valueOf(estimatedDuration)
+				).build()));
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			_mvcActionCommand.processAction(
+				mockLiferayPortletActionRequest,
+				mockLiferayPortletActionResponse);
+
+			MockHttpServletResponse mockHttpServletResponse =
+				(MockHttpServletResponse)
+					mockLiferayPortletActionResponse.getHttpServletResponse();
+
+			JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+				mockHttpServletResponse.getContentAsString());
+
+			Assert.assertEquals(
+				estimatedDuration,
+				jsonObject.get("segmentsExperimentEstimatedDaysDuration"));
+		}
+	}
+
+	private SegmentsExperience _addSegmentsExperience() throws Exception {
+		return _segmentsExperienceLocalService.addSegmentsExperience(
+			TestPropsValues.getUserId(), _group.getGroupId(), 0,
+			_portal.getClassNameId(Layout.class), _layout.getPlid(),
+			Collections.singletonMap(LocaleUtil.getSiteDefault(), "Variant 1"),
+			false, new UnicodeProperties(true), _serviceContext);
+	}
+
+	private SegmentsExperiment _addSegmentsExperiment() throws Exception {
```

Comment:
What do you think about calling SegmentsTestUtil.addSegmentsExperiment?

Reply:
Solved. Thanks